regenerate.py

```diff
@@ -4,77 +4,64 @@
 
 import argparse
 import glob
-import itertools
 import logging
 import os
 import subprocess
 
 logger = logging.getLogger(__name__)
 LOGGING_LEVELS = {
-  'NOTSET': logging.NOTSET,
-  'DEBUG': logging.DEBUG,
-  'INFO': logging.INFO,
-  'WARNING': logging.WARNING,
-  'ERROR': logging.ERROR,
-  'CRITICAL': logging.CRITICAL,
+    'NOTSET': logging.NOTSET,
+    'DEBUG': logging.DEBUG,
+    'INFO': logging.INFO,
+    'WARNING': logging.WARNING,
+    'ERROR': logging.ERROR,
+    'CRITICAL': logging.CRITICAL,
 }
 DEFAULT_PANDOC_FLAGS = [
-  '-s',
-  '-H',
-  'header.html',
-  '-B',
-  'title.html',
+    '-s',
+    '--template=template.html',
+    '--base-header-level=2',
+    '--katex',
 ]
 
 
 def run(*args) -> str:
   logging.debug('Running command: %s', args)
-  result = subprocess.run(
-    args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding='utf-8')
+  result = subprocess.run(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding='utf-8')
   logging.debug('\nStdout: %s\nStderr: %s', result.stdout, result.stderr)
   return result
 
 
-def GenerateHtml(pandoc_flags: List[str], doc_paths: List[str],
-                 post_paths: List[str]):
+def GenerateHtml(pandoc_flags: List[str], doc_paths: List[str], post_paths: List[str],
+                 build_dir: str):
   # TODO: Process multiple files concurrently.
   def output(input_path, output_path):
     logging.info('Reading from %s\nWriting to %s', input_path, output_path)
-    args = (['pandoc'] + DEFAULT_PANDOC_FLAGS + pandoc_flags +
-            [input_path, '-o', output_path])
+    args = (['pandoc'] + DEFAULT_PANDOC_FLAGS + pandoc_flags + [input_path, '-o', output_path])
     run(*args)
 
   for post_path in post_paths:
-    output_dir = os.path.splitext(post_path)[0]
+    output_dir = os.path.join(build_dir, os.path.splitext(post_path)[0])
```

Comment:
Since we're using a build directory now, should the `os.mkdir` call become `os.mkdirs`?

Reply:
To ensure the build directory exists? I suppose so, yeah. Will change.